File: stock-portfolio-app/src/app/stock.service.ts
import {Injectable} from '@angular/core';
import {Observable} from "rxjs";
import {HttpClient} from "@angular/common/http";
import {StockSearch} from "./stock-search/stockSearch";
import {MarketInsight} from "./stocks/marketInsight";
import {CompanyDetails} from "./add-stock/companyDetails";
import {StockDataLatest} from "./add-stock/stock-data-latest";

@Injectable({
  providedIn: 'root'
})
export class StockService {

  private apiUrl = 'http://localhost:8080/';

  constructor(private http: HttpClient) {
  }

  searchStock(name: string): Observable<StockSearch> {
    let getUrl = this.apiUrl + "stock/search?symbol=" + name;
    return this.http.get<StockSearch>(getUrl);
  }

  marketInsight(): Observable<MarketInsight> {
    let getUrl = this.apiUrl + "insight";
    return this.http.get<MarketInsight>(getUrl);
  }

  getStockDataLatest(name: string): Observable<StockDataLatest> {

    let getUrl = this.apiUrl + "stock?symbol=" + name;

    return this.http.get<StockDataLatest>(getUrl);
  }

  getCompanyOverview(name: string): Observable<CompanyDetails> {
    let getUrl = this.apiUrl + "company/detail?symbol=" + name;
    return this.http.get<CompanyDetails>(getUrl);
  }
}

File: portfolio-rest-api/src/main/java/com/bank/stock/portfolio/service/StockService.java
package com.bank.stock.portfolio.service;

import com.bank.stock.portfolio.service.bos.CompanyDetailsBO;
import com.bank.stock.portfolio.service.bos.MarketInsightBO;
import com.bank.stock.portfolio.service.bos.StockDataBO;
import com.bank.stock.portfolio.service.bos.StockSearchBO;

public interface StockService {

    StockSearchBO searchStock(String symbol);
    MarketInsightBO marketInsight();
    StockDataBO getStockData(String symbol);
    CompanyDetailsBO getCompanyOverview(String symbol);

}
